### PR TITLE
Condition

### PR DESCRIPTION
**Pull request by baiju-grh**
Add condition to fhir transformer

### CODE REVIEW

File: Microsoft.HealthVault.Fhir/Constants/HealthVaultVocabularies.cs [hunk 1]
```diff
@@ -27,6 +27,9 @@ public static class HealthVaultVocabularies
         public const string BodyCompositionSites = "body-composition-sites";
         public const string BodyCompositionMeasurementNames = "body-composition-measurement-names";
 
+        public const string ConditionOccurence = "condition-occurence";
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Constants/HealthVaultVocabularies.cs]: >ConditionOccurence [](start = 28, length = 18)

Typo, Occurrence (Double C and R) Please fix everywhere in this file.

File: Microsoft.HealthVault.Fhir/FhirExtensions/DomainResourceExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    public static class DomainResourceExtensions
+    {
+        public static void AddNoteAsText(this DomainResource domainResource, string note)
+        {
+            if (!string.IsNullOrEmpty(note))
+            {
+                domainResource.Text = new Narrative() { Div = note };
+            }
+        }
+    }
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/DomainResourceExtensions.cs]: When you merge with Jeffy's check in make sure you include the copyright header

File: Microsoft.HealthVault.Fhir/FhirExtensions/ExtensionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        {
+            return extendable.AddStringAsExtension(HealthVaultVocabularies.FlagsFhirExtensionName, flags.ToString());
+        }
+
+        public static Extension AddStateAsExtension(this IExtendable extendable, ThingState state)
+        {
+            return extendable.AddStringAsExtension(HealthVaultVocabularies.StateFhirExtensionName, state.ToString());
+        }
+    }
+}
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/ExtensionExtensions.cs]: When you merge with Jeffy's check in make sure you include the copyright header

File: Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using Hl7.Fhir.Model;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs]: Add file header.

File: Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    public static class FhirDateTimeExtensions
+    {
+        public enum FhirDateTimePrecision
+        {
+            Year = 4,       //2017
+            Month = 7,      //2017-22
+            Day = 10,       //2017-22-21
+            Minute = 15,    //2017-22-21T13:45
+            Second = 18    //2017-22-21T13:45:21
+        }
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs]: This looks really cryptic, please add comments on what this enum's purpose is. Are you using an enum to count characters?

File: Microsoft.HealthVault.Fhir/Transformers/ApproximateDateTimeToFhir.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        public static FhirDateTime ToFhir(this ApproximateDateTime approximateDateTime)
+        {
+            return ApproximateDateTimeToFhir.ToFhirInternal(approximateDateTime);
+        }
+    }
+    internal static class ApproximateDateTimeToFhir
+    {
+        internal static FhirDateTime ToFhirInternal(ApproximateDateTime approximateDateTime)
+        {
+            if (!approximateDateTime.ApproximateDate.Month.HasValue && !approximateDateTime.ApproximateDate.Day.HasValue)
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/ApproximateDateTimeToFhir.cs]: Add open and closing curly braces everywhere on this file. Even when we have only one statement the guidance is to open and close braces for readibility.

File: Microsoft.HealthVault.Fhir/Transformers/ApproximateDateTimeToFhir.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+                return new FhirDateTime(approximateDateTime.ApproximateDate.Year);
+            else if (approximateDateTime.ApproximateDate.Month.HasValue && !approximateDateTime.ApproximateDate.Day.HasValue)
+                return new FhirDateTime(approximateDateTime.ApproximateDate.Year, approximateDateTime.ApproximateDate.Month.Value);
+            else if (approximateDateTime.ApproximateDate.Month.HasValue && approximateDateTime.ApproximateDate.Day.HasValue && (approximateDateTime.ApproximateTime == null || !approximateDateTime.ApproximateTime.HasValue))
+                return new FhirDateTime(approximateDateTime.ApproximateDate.Year, approximateDateTime.ApproximateDate.Month.Value, approximateDateTime.ApproximateDate.Day.Value);
+            else
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/ApproximateDateTimeToFhir.cs]: Can you find a way to simplify these if/else statements it is hard to follow. 

You could probably group them

if(!day){ 
     if(!month){

     } else { 

     }
}

and so on, to make it easy to follow and make sure we do not miss a case when we update it.

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 8]
```diff
@@ -0,0 +1,154 @@
+﻿using System;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.Fhir.FhirExtensions;
+using Microsoft.HealthVault.Thing;
+using HVCondition = Microsoft.HealthVault.ItemTypes.Condition;
+
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: Add Header

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class ConditionToHealthVault
+    {
+        public static HVCondition ToHealthVault(this Condition fhircondition)
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: >fhircondition [](start = 63, length = 13)

Casing is wrong, please use camelCasing fhirCondition

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+        {
+            return fhircondition.ToCondition();
+        }
+    }
+    internal static class FhirConditionToHealthVaultCondition
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: add an extra line between the bracket and the internal keyword

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+                    var value = code.Code.Split(':');
+                    var vocabCode = value.Length == 2 ? value[1] : null;
+                    var vocab = value.Length == 2 ? value[0] : HealthVaultVocabularies.Fhir;
+
+                    hvCondition.SetConditionName(
+                            code.Display,
+                            vocabCode,
+                            vocab,
+                            code.System,
+                            code.Version);
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: >                     [](start = 42, length = 20)

remove extra spaces

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+            {
+                foreach (Annotation annotation in fhircondition.Note)
+                {
+                    hvCondition.CommonData.Note += annotation.Text;
+                }                
+            }
+            
+            return hvCondition;
+        }
+        private static ItemTypes.ApproximateDateTime GetHealthVaultApproximateTimeFromEffectiveDate(Element fhirElement)
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: >GetHealthVaultApproximateTimeFromEffectiveDate [](start = 53, length = 46)

This function doesn't belong here, it looks too generic to be only related to Condition, please move out

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+        {
+            FhirDateTime fhirdateTime = null;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: >fhirdateTime [](start = 25, length = 12)

camel casing

File: Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+namespace Microsoft.HealthVault.Fhir.FhirExtensions
+{
+    public static class FhirDateTimeExtensions
+    {
+        public enum FhirDateTimePrecision
+        {
+            Year = 4,       //2017
+            Month = 7,      //2017-22
+            Day = 10,       //2017-22-21
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs]: >Day = 10,       //2017-22-21 [](start = 12, length = 28)

What happens with 2017-7-7 or 2017-7-31?

File: Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+using Hl7.Fhir.Model;
+
+namespace Microsoft.HealthVault.Fhir.FhirExtensions
+{
+    public static class FhirDateTimeExtensions
+    {
+        public enum FhirDateTimePrecision
+        {
+            Year = 4,       //2017
+            Month = 7,      //2017-22
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs]: >//2017-22 [](start = 28, length = 9)

22 is not a valid month, fix the comment

File: Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+            Day = 10,       //2017-22-21
+            Minute = 15,    //2017-22-21T13:45
+            Second = 18    //2017-22-21T13:45:21
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs]: >Second = 18    //2017-22-21T13:45:21 [](start = 12, length = 36)

Could we receive dates with time zones? What about 2017-01-01T13:00:00Z?

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 17]
```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Codings;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.Fhir.FhirExtensions;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: Add Header

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+    }
+    /// <summary>
+    /// An extension class that transforms HealthVault condition data types into FHIR Condition
+    /// </summary>
+    internal static class HealthVaultConditionToFhir
+    {
+
+        internal static Condition ToFhirInternal(HVCondition cd, Condition fhirCondition)
+        {
+
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: remove extra line

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+                var note = new Annotation();
+                note.Text = cd.CommonData.Note;
+                fhirCondition.Note = new List<Annotation> {note};
+                fhirCondition.AddExtension(HealthVaultVocabularies.ConditionSource, new FhirString(cd.CommonData.Source));
+            }
+
+            if (cd.Name != null)
+            {
+                fhirCondition.Code = new CodeableConcept() { Coding = HealthVaultCodesToFhir.ConvertCodableValueToFhir(cd.Name, fhirCodes) };
+                fhirCondition.Code.Text = fhirCondition.Code.Coding[0].Display;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: > fhirCondition.Code.Text = fhirCondition.Code.Coding[0].Display; [](start = 15, length = 64)

Why repeat the display in the coding? Should we assign something else?

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+                if (cvalue.Value != null)
+                {
+                    if (Enum.TryParse(cvalue.Value, true, out clinicalStatusCode))
+                    {
+                        fhirCondition.ClinicalStatus = clinicalStatusCode;
+                        return;
+                    }
+                    else
+                    {
+                        fhirCondition.AddExtension(HealthVaultVocabularies.ConditionOccurenceExentsionName, new FhirString(cvalue.Value));
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: >ConditionOccurenceExentsionName [](start = 75, length = 31)

Typo Extension*

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+            {
+                fhirCondition.AddExtension(HealthVaultVocabularies.ConditionStopReason, new FhirString(cd.StopReason));
+            }
+
+            return fhirCondition;
+        }
+
+        private static void SetClinicalStatusCode(this Condition fhirCondition, ItemTypes.CodableValue status)
+        {
+            foreach (ItemTypes.CodedValue cvalue in status)
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: >cvalue [](start = 42, length = 6)

Came case

File: Microsoft.HealthVault.Fhir.UnitTests/ToFhirTests/ConditionToFhirTest.cs [hunk 22]
```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Transformers;
+using Microsoft.HealthVault.Thing;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using HVCondition = Microsoft.HealthVault.ItemTypes.Condition;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/ToFhirTests/ConditionToFhirTest.cs]: header

File: Microsoft.HealthVault.Fhir.UnitTests/ToFhirTests/ConditionToFhirTest.cs [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+namespace Microsoft.HealthVault.Fhir.UnitTests.ToFhirTests
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/ToFhirTests/ConditionToFhirTest.cs]: add extra line between namespace and using statements

File: Microsoft.HealthVault.Fhir.UnitTests/ToHealthVaultTests/ConditionToHealthVaultCondition.cs [hunk 24]
```diff
@@ -0,0 +1,32 @@
+﻿using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Microsoft.HealthVault.Fhir.Transformers;
+using Microsoft.HealthVault.Fhir.UnitTests.Samples;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/ToHealthVaultTests/ConditionToHealthVaultCondition.cs]: header

File: Microsoft.HealthVault.Fhir.UnitTests/ToHealthVaultTests/ConditionToHealthVaultCondition.cs [hunk 25]
```diff
@@ -0,0 +1,32 @@
...
+            Assert.AreEqual("Acute renal insufficiency specified as due to procedure", cd.Name.Text);
+            Assert.AreEqual(new HVItemTypes.ApproximateDateTime() {ApproximateDate = new HVItemTypes.ApproximateDate(2013, 03, 11)}, cd.OnsetDate);
+            Assert.AreEqual("In Control", cd.StopReason);
+            Assert.AreEqual("intermittent", cd.Status.Text);
+            Assert.AreEqual("The patient is anuric.The patient state is critcal.",cd.CommonData.Note);
+            Assert.AreEqual(new HVItemTypes.ApproximateDateTime() { ApproximateDate = new HVItemTypes.ApproximateDate(2015)}, cd.StopDate);
+            Assert.AreEqual("patient via InstantPHR patient portal", cd.CommonData.Source);
+
+        }
+
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/ToHealthVaultTests/ConditionToHealthVaultCondition.cs]: remove line

File: Microsoft.HealthVault.Fhir.UnitTests/ToHealthVaultTests/ConditionToHealthVaultCondition.cs [hunk 26]
```diff
@@ -0,0 +1,32 @@
...
+{
+    [TestClass]
+    public class ConditionToHealthVaultCondition
+    {
+        [TestMethod]
+        public void WhenConditionToHealthVault_ThenValuesEqual()
+        {
+            var json = SampleUtil.GetSampleContent("FhirCondition.json");
+            var fhirParser = new FhirJsonParser();
+            var fhircondition = fhirParser.Parse<Condition>(json);
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/ToHealthVaultTests/ConditionToHealthVaultCondition.cs]: >fhircondition [](start = 16, length = 13)

camel casing

File: Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs (hunk 14 above)
Comment (baiju-grh) [inline Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs]: FhirDateTime expects in the format 2017-07-28
As per regular expression of FhirDateTime, month and date should have 2 digits 
( https://www.hl7.org/fhir/datatypes.html#dateTime )

**Comment by brandonpollett** [comment]
@baiju-grh can you bring this up to date with master?

File: Microsoft.HealthVault.Fhir/Constants/HealthVaultVocabularies.cs [hunk 27]
```diff
@@ -52,7 +55,12 @@ public static class HealthVaultVocabularies
         public const string SleepJournalAMMedication = "sleep-journal-medication";
 
         public const string IrregularHeartBeatExtensionName = BaseUri + "vital-signs/blood-pressure/irregular-heartbeat";
-     
+
+        public const string ConditionOccurrenceExtensionName = BaseUri + ConditionOccurrence;
```
Comment (brandonpollett) [inline Microsoft.HealthVault.Fhir/Constants/HealthVaultVocabularies.cs]: Since these are extensions, can you move them to Microsoft.HealthVault.Fhir/Constants/HealthVaultExtensions.cs and use the URL specified there?

Comment (brandonpollett) [reply Microsoft.HealthVault.Fhir/Constants/HealthVaultVocabularies.cs]: We've also been adding the structure definition files in the folder: `Microsoft.HealthVault.Fhir/FhirExtensions/XmlDefinitions`. You can generate it using the tool Forge from Furore.

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 28]
```diff
@@ -0,0 +1,107 @@
...
+        }
+    }
+    /// <summary>
+    /// An extension class that transforms HealthVault condition data types into FHIR Condition
+    /// </summary>
+    internal static class HealthVaultConditionToFhir
+    {
+        internal static Condition ToFhirInternal(HVCondition cd, Condition fhirCondition)
+        {
+            var fhirCodes = new List<Coding>();
```
Comment (brandonpollett) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: This doesn't look like it's used anymore.

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+            }
+            else
+            {
+                hvCondition.Status = new ItemTypes.CodableValue(fhirCondition.GetStringExtension(HealthVaultVocabularies.ConditionOccurrenceExtensionName));
+                hvCondition.Status.Add(new ItemTypes.CodedValue(fhirCondition.GetStringExtension
+                    (HealthVaultVocabularies.ConditionOccurrenceExtensionName), HealthVaultVocabularies.ConditionOccurrence, HealthVaultVocabularies.Wc, "1"));
+            }
+
+            if (fhirCondition.Code.IsNullOrEmpty())
+                throw new System.ArgumentNullException($"Can not transform a {typeof(Condition)} with no code into {typeof(HVCondition)}");
```
Comment (brandonpollett) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: Why not just make this a string instead of doing the string interpolation?

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs (hunk 8 above)
Comment (baiju-grh) [reply Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: Header added to file

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs (hunk 17 above)
Comment (baiju-grh) [reply Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: Header added to file.

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs (hunk 29 above)
Comment (vsmahesh) [reply Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: Since we are transforming from one Condition to the other, it is worth mentioning the namespace. Because an external type is referred, it is better to construct the message dynamically rather than hard-coding it.

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+
+            string conditionOccurence = string.Empty;
+            var conditionExtension = fhirCondition.GetExtension(HealthVaultExtensions.Condition);
+            if (conditionExtension != null)
+            {
+                hvCondition.StopReason = conditionExtension.GetStringExtension(HealthVaultExtensions.ConditionStopReason);
+                conditionOccurence = conditionExtension.GetStringExtension(HealthVaultExtensions.ConditionOccurrence);
+            }
+
+
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: nit: remove extra line

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 31]
```diff
@@ -0,0 +1,64 @@
...
+            hvCondition.OnsetDate = fhirCondition.Onset.ToAproximateDateTime();
+            hvCondition.StopDate = fhirCondition.Abatement.ToAproximateDateTime();
+
+
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: nit: remove extra line

File: Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirCondition.json [hunk 32]
```diff
@@ -0,0 +1,113 @@
...
+  },
+  "extension": [{
+		"url": "http://healthvault.com/fhir-extensions/thing-flags",
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir.UnitTests/Samples/FhirCondition.json]: >		 [](start = 0, length = 2)

clean up tabs, turn them into spaces for consistency around the repo. You can do this easily in visual studio.

File: Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs [hunk 33]
```diff
@@ -0,0 +1,110 @@
...
+
+            if (cd.CommonData != null)
+            {
+                fhirCondition.AddNoteAsText(cd.CommonData.Note);
+                var note = new Annotation();
+                note.Text = cd.CommonData.Note;
+                fhirCondition.Note = new List<Annotation> {note};
+            }
+                    
+            fhirCondition.Code = cd.Name.ToFhir();
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/HealthVaultConditionToFhir.cs]: >Name [](start = 36, length = 4)

Could name be null? Do a null check

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 34]
```diff
@@ -0,0 +1,64 @@
...
+            }
+            else if (!string.IsNullOrWhiteSpace(conditionOccurence))
+            {
+                hvCondition.Status = new ItemTypes.CodableValue(conditionOccurence);
+                hvCondition.Status.Add(new ItemTypes.CodedValue(conditionOccurence, HealthVaultVocabularies.ConditionOccurrence, HealthVaultVocabularies.Wc, "1"));
+            }
+
+            if (fhirCondition.Code.IsNullOrEmpty())
+                throw new System.ArgumentNullException($"Can not transform a {typeof(Condition)} with no code into {typeof(HVCondition)}");
+
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: add open and close curly braces.

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+                hvCondition.Status = new ItemTypes.CodableValue(fhirCondition.ClinicalStatus.Value.ToString());
+                hvCondition.Status.Add(new ItemTypes.CodedValue(fhirCondition.ClinicalStatus.Value.ToString(), FhirCategories.Hl7Condition, HealthVaultVocabularies.Fhir, ""));
+            }
+            else if (!string.IsNullOrWhiteSpace(conditionOccurence))
+            {
+                hvCondition.Status = new ItemTypes.CodableValue(conditionOccurence);
+                hvCondition.Status.Add(new ItemTypes.CodedValue(conditionOccurence, HealthVaultVocabularies.ConditionOccurrence, HealthVaultVocabularies.Wc, "1"));
+            }
+
+            if (fhirCondition.Code.IsNullOrEmpty())
```
Comment (gcorvera) [inline Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: >if (fhirCondition.Code.IsNullOrEmpty()) [](start = 12, length = 39)

why do we do this check this late? Move this code up. I am more inclined to return null than throw an exception. If we do not support a type, when transforming a bundle why would we like to fail the whole transformation instead of just skipping the ones we do not understand?

Comment (jeffy-grh) [reply Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: @gcorvera Returning a null will leave the developer in darkness about what went wrong. 
@baiju-grh `NotSupportedException` would be better here
@jeffy-grh We need to standardize and document possible exceptions.

File: Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs (hunk 31 above)
Comment (baiju-grh) [reply Microsoft.HealthVault.Fhir/Transformers/FhirConditionToHealthVaultCondition.cs]: removed extra line
